site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "print('shape: ', repr(path_ds.output_shapes))\n",
+        "print('type: ', path_ds.output_types)\n",
+        "print()\n",
+        "print(path_ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "ZjyGcM8OwBJ2",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Now create a new dataset that loads and formats images on the fly by mapping `preprocess_image` over the dataset of paths."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "D1iba6f4khu-",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "image_ds = path_ds.map(load_and_preprocess_image, num_parallel_calls=8)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "JLUPs2a-lEEJ",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import matplotlib.pyplot as plt\n",
+        "\n",
+        "plt.figure(figsize=(8,8))\n",
+        "for n,image in enumerate(image_ds.take(4)):\n",
+        "  plt.subplot(2,2,n+1)\n",
+        "  plt.imshow(image)\n",
+        "  plt.grid(False)\n",
+        "  plt.xticks([])\n",
+        "  plt.yticks([])\n",
+        "  plt.xlabel(caption_image(all_image_paths[n]))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "P6FNqPbxkbdx",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### A dataset of `(image, label)` pairs"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "YgvrWLKG67-x",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Using the same `from_tensor_slices` method we can build a dataset of labels"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "AgBsAiV06udj",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "label_ds = tf.data.Dataset.from_tensor_slices(tf.cast(all_image_labels, tf.int64))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "HEsk5nN0vyeX",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "for label in label_ds.take(10):\n",
+        "  print(label_names[label.numpy()])"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "jHjgrEeTxyYz",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Since the datasets are in the same order we can just zip them together to get a dataset of `(image, label)` pairs."
```

Comment:
Would it be worth mentioning that you can also call `from_tensor_slices()` with a pair of tensors directly?

Reply:
I only did it this way because we had the images in a Dataset already.

I've added an example to the end of this section demonstrating:

```
ds = tf.data.Dataset.from_tensor_slices((all_image_paths, all_image_labels))

# The tuple elements are unpacked into the positional arguments of the mapped function 
def load_and_preprocess_from_path_label(path, label):
  return load_and_preprocess_image(path), label

image_label_ds = ds.map(load_and_preprocess_from_path_label)
image_label_ds
```